### PR TITLE
refactor: indicate if saving has been cancelled in `askUserToSaveFile`

### PR DESCRIPTION
**Pull request by Panzenbaby**
Resolves https://github.com/PayvoHQ/wallet/issues/556
Resolves https://github.com/PayvoHQ/wallet/issues/557

## Summary

Adjust the FileSystemPluginService so that it corresponds to the signature provided by PluginAPI.filesystem. Now a suggested file name can be passed by developers.

I couldn't tell whether a user canceled the dialog or really saved the file. So I avoided treating showSaveDialog cancellation as a success and throw an error instead.

## Checklist
-   [ x ] I checked my changes for obvious issues, debug statements and commented code
-   [ x ] Tests _(if necessary)_
-   [ x ] Ready to be merged

I adjusted one test but the argument doesn't seem to be required. I added it anyway to match the signature and remove the lint error.

### CODE REVIEW

**Comment by Panzenbaby** [comment]
It seems that the action can not checkout my branch to run the checks :(

File: src/plugins/services/filesystem/FileSystemPluginService.ts [hunk 1]
```diff
@@ -17,8 +17,12 @@ export class FileSystemPluginService implements PluginService {
 		};
 	}
 
-	private async askUserToSaveFile(content: string) {
-		const { filePath } = await electron.remote.dialog.showSaveDialog({});
+	private async askUserToSaveFile(content: string, suggestedFileName?: string) {
+		const { canceled, filePath } = await electron.remote.dialog.showSaveDialog({defaultPath: suggestedFileName});
+
+		if (canceled) {
+			throw new Error('Canceled by user');
```
Comment (faustbrian) [inline src/plugins/services/filesystem/FileSystemPluginService.ts:24]: What's the reason for this having to throw an error instead of returning early?

**Review by faustbrian — CHANGES_REQUESTED** [review]
Missing tests

File: src/plugins/services/filesystem/FileSystemPluginService.ts (hunk 1 above)
Comment (Panzenbaby) [reply src/plugins/services/filesystem/FileSystemPluginService.ts:24]: If you exit this method in the same way in each case, the caller will not be able to determine whether the save was successful or not.
Another solution would be to add a return type like boolean that indicates the success of this operation.

Comment (Panzenbaby) [reply src/plugins/services/filesystem/FileSystemPluginService.ts]: I changed the behavior to return a boolean.

**Comment by Panzenbaby** [comment]
@faustbrian 
There is no button to resolve your requested changes but I added tests. Could you please review again.

**Comment by goga-m** [comment]
@Panzenbaby is there a particular plugin you are using that needs this change? Can you provide an example so that we could reproduce it in the wallet? 

Thanks in advance

**Comment by Panzenbaby** [comment]
@goga-m I'm about to rewrite https://github.com/Panzenbaby/ArkStakingRewardReport for the payvo wallet. The plugin offers an export function and I want to display a success or error message if the file was saved or an error occurred. In the event that the user cancels the save dialog, I get the same result as if they had successfully saved the file.

You can check the current behavior by calling
`api.filesystem().askUserToSaveFile('test', 'test').then(() => console.log('I can't tell if the file was saved or if the user abort the dialog')).catch((error) => console.log('An error occurred'));`

Try this once with selecting save and once with cancel. Both leads to the same result.

**Pull request merged**